### ISSUE TITLE
Investigate Publishing to An App Store

### ISSUE THREAD

**Issue by johnbaku** [fetlife/android#179]
Off of a suggestion from a member, I checked out the Amazon Android store and it looks like we might be able to get into it.

We should do some research and if so consider publishing the app in the store.

Thoughts?

**Comment by FernandoMiguel** [comment]
I have no problem with it. 
But how many users have the Amazon store app installed? 
And to have it installed on stock Android, you still require unknown sources to be enabled. 
That leaves Amazon devices with it preinstalled

**Comment by thezerogod** [comment]
If you're going to consider Amazon, you may also want to see if you can
distribute via F-Droid. Neither is generally preinstalled on Android phones
but both work roughly the same way, and for those of us that would prefer
not to have the Amazon store on (or who are otherwise trying to focus on
open source software) it would be nice.

Either distribution channel should also allow users to turn off installing
from unknown sources after the initial app-store install, rather than
leaving it off or toggling for every app update.

But for what it's worth, neither of those really appeals to me - I prefer
to install from GitHub.

**Comment by malachid** [comment]
They won't allow any apps that link to fetlife.com
https://developer.amazon.com/appsandservices/support/faq#policy_guidelines

**Comment by johnbaku** [comment]
@malachid I've heard through the grapevine they allow swinger sites that have nudity in them as well. I wonder if it's a case of that is what they state... but not what they do?

**Comment by GojuSuzi** [comment]
More likely it's a case of they don't bother enforcing unless it's reported, and with FL's higher propensity for drama llamas than those other sites (through sheer user base volume), that risk is increased.

Yes, it'd be good, but there's too big a risk of it getting pulled and then folk not realising that updates are only being shared here/elsewhere instead; just look at the number of folk commenting even recently on the unofficial app's group that their app is not working despite announcements forever ago. People have been trained to be lazy with app updates, they just happen, so they do not actively check for any issues with that process unless the app becomes unusable or they switch to a new device. Better to stick to a stable source like this.

**Comment by kistitan** [comment]
I wonder how the swinger all went through as this seems to very straightforward:

"Pornography: We prohibit apps containing pornography or that depict graphic sexual acts or sexually explicit material. This includes full nudity contained in your application or marketing materials. We also don’t allow content that drives traffic to pornography sites."

Maybe the app has pre-aproval policy for pictures.
Like Whiplr has.
Btw Whiplr is still allowed to stay in app stores.
They recently had to remove several user photos and content but after they got green light AFAIK.

**Comment by kistitan** [comment]
Unfortunately I cannot download the swinger app to check how far can we go.
Anybody with different location can take a look at it?

However unfortunately this seems to be pretty clear to me:
"Pornography: We prohibit apps containing pornography or that depict graphic sexual acts or sexually explicit material. This includes full nudity contained in your application or marketing materials. We also don’t allow content that drives traffic to pornography sites."

@johnbaku 
What do you think? Should we still keep this topic open? Or maybe change it to investigation about MiKandi?

**Comment by johnbaku** [comment]
@kistitan I've posted the following question as a status update on my profile:

> Hey FetLife, have you personally downloaded any adult oriented apps from any of the Android stores? If so, mind sharing with me the name of the store and app? 

Ref: https://fetlife.com/users/1/statuses/20441643

It's possible Amazon says that but doesn't adhere to that.

Can anyone here download this app and maybe see if they openly have adult material?
https://www.amazon.ca/DBH-SLS/dp/B00E6QGVQQ

**Comment by malachid** [comment]
When we were on the Google Play Store, we were very visible (top 10). OEMs (including Google) reached out to us for permission to test it on various hardware. After their policy change, they pulled it without notice and told us the only way we could republish was if we no longer connected to fetlife.com (that's what they were checking for).  They said that, if we tried to republish, they would delete our Google account.

Other apps were (are) still there - but it is probably just a matter of time.  We had voluntarily selected "adult only" when we published the app, and as I said were very visible.  Their policy change came when they announced Google Play for Education and started trying to make the store safer for kids.

We looked at alternate stores, like Amazon, but they had similar policies.  I can't honestly remember if we reached out to them or not.

We had reached out to Stripe to find out whether we could make it donation based instead.  They stated that they wanted to work with us, but their current agreements with Visa and Mastercard would not allow them.

MiKandi reached out to us and hosted us; but our user base did not like that hosting solution very much.

**Comment by johnbaku** [comment]
@malachid I could be wrong but in some cases I think having a free app might change things a bit. Because all of a sudden the agreements that these stores have with the banks they are dealing with won't be an issue.

**Comment by johnbaku** [comment]
It seems like Amazon does have an Adult rating and according to there Adult rating it sould like we would be fine:

![sls__amazon_ca__appstore_for_android](https://cloud.githubusercontent.com/assets/22100/19045490/ebb7ec38-894d-11e6-977a-bc2e52129f18.png)

It can't hurt to try no?

**Comment by FernandoMiguel** [comment]
it wouldnt be bad to try amazon. just keep in mind to NOT use a main account.
if they are like google and decide to kill it, you will lose ALL of amazon services from there, and all funds

**Comment by FernandoMiguel** [comment]
@johnbaku based on your post on fet, i dont see many people even understanding what a App Store is... we might have loads of trouble

**Comment by johnbaku** [comment]
@FernandoMiguel LOL... yeah... I saw that! :-p

**Comment by nuntius35** [comment]
If certain technical requirements are fulfilled and if the whole app is truely free and open source, than submission to F-Droid should be really easy, see https://f-droid.org/forums/topic/read-this-before-posting/ 
It makes life much easier if you can get apps via an app store.

**Comment by johnbaku** [comment]
@nuntius35 Never knew about F-Droid... cool! That sounds like an interesting option as well!

**Comment by gkirilov** [comment]
You may also want to consider the option of the app itself checking for updates.
Dropbox, for example, does that - they have a "Get early releases" checkbox in the settings menu.
They only use it for development versions, but you can use it for all releases.
Even if the initial install has been through github, at least it makes it much easier to keep up with updates.

**Comment by kistitan** [comment]
@gkirilov 
Thanks for the suggestion.
The issue with the update check that it should be executed periodically so the app should either run a continious background process or wake itself up periodically.
It is not bad thing at all, we might going to add anyway, but some user tend to complain about these things as they might drain more battery and they might look suspicious. 
Also some background app blocker can block them before the we can check for updates.

So our strategy now is that we send out push notification after every new release with the download link, so instead of polling, update notification will be pushed to the user's phone.

That gives us a little more work after a release (until it is not automated) but it saves an extra background process and connection on all the user's devices.

**Comment by reneklacan** [comment]
@kistitan what about just having a simple check when you open an app instead of having background process? Then app would still be auto-updatable without push notification and it wouldn't rely on background process, just on the fact that user opens an app from time to time.

**Comment by kistitan** [comment]
Some note from our side about F-Droid here: 
https://github.com/fetlife/android/issues/249

**Comment by kistitan** [comment]
I just noticed I did not reply on @reneklacan comment.

This approach seems to be good, however there should be also a rule that we apply, not bothering always the users with a pop-up at every app start if (s)he wants to download the update.

But maybe a (semi-)persistent notification in the notification area might work.
Or maybe rather a marker somewhere in the menu next the the about marker.

Yes I like the approach. I will open a new issue about it.
